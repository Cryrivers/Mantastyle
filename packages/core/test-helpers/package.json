{
  "name": "@manta-style/test-helpers",
  "version": "0.2.0-alpha.9",
  "main": "src/index.ts",
  "dependencies": {
    "prettier": "^1.14.3"
  },
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "@types/prettier": "^1.13.2"
  }
}
