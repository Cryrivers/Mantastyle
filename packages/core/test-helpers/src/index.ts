import * as BuilderPluginTestHelper from './builder-tester';
export { BuilderPluginTestHelper };
