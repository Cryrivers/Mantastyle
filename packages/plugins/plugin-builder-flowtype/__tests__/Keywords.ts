import getTranspiledString from '../src/utils/transpile';
import { BuilderPluginTestHelper } from '@manta-style/test-helpers';

const testTranspiledString = BuilderPluginTestHelper.testTranspiledString(
  getTranspiledString,
);

describe('Keywords', () => {
  test('AnyKeyword', () => {
    expect(testTranspiledString('type Test = any;')).toMatchSnapshot();
  });
  test('NumberKeyword', () => {
    expect(testTranspiledString('type Test = number;')).toMatchSnapshot();
  });
  test('BooleanKeyword', () => {
    expect(testTranspiledString('type Test = boolean;')).toMatchSnapshot();
  });
  test('NeverKeyword', () => {
    expect(testTranspiledString('type Test = void;')).toMatchSnapshot();
  });
  test('ObjectKeyword', () => {
    expect(testTranspiledString('type Test = Object;')).toMatchSnapshot();
  });
  test('StringKeyword', () => {
    expect(testTranspiledString('type Test = string;')).toMatchSnapshot();
  });
  test('UndefinedKeyword', () => {
    expect(testTranspiledString('type Test = undefined;')).toMatchSnapshot();
  });
  test('EmptyKeyword', () => {
    expect(testTranspiledString('type Test = empty;')).toMatchSnapshot();
  });
  test('NullKeyword', () => {
    expect(testTranspiledString('type Test = null;')).toMatchSnapshot();
  });
});
