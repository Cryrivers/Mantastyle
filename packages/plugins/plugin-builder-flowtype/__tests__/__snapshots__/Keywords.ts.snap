// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keywords AnyKeyword 1`] = `
"import _Runtime from \\"@manta-style/runtime\\";

const Test = _Runtime.TypeAliasDeclaration(\\"Test\\", function (typeFactory) {
  const type = _Runtime.AnyKeyword;
  return type;
}, []);"
`;

exports[`Keywords BooleanKeyword 1`] = `
"import _Runtime from \\"@manta-style/runtime\\";

const Test = _Runtime.TypeAliasDeclaration(\\"Test\\", function (typeFactory) {
  const type = _Runtime.BooleanKeyword;
  return type;
}, []);"
`;

exports[`Keywords EmptyKeyword 1`] = `
"import _Runtime from \\"@manta-style/runtime\\";

const Test = _Runtime.TypeAliasDeclaration(\\"Test\\", function (typeFactory) {
  const type = _Runtime.NeverKeyword;
  return type;
}, []);"
`;

exports[`Keywords NeverKeyword 1`] = `
"import _Runtime from \\"@manta-style/runtime\\";

const Test = _Runtime.TypeAliasDeclaration(\\"Test\\", function (typeFactory) {
  const type = _Runtime.UndefinedKeyword;
  return type;
}, []);"
`;

exports[`Keywords NullKeyword 1`] = `
"import _Runtime from \\"@manta-style/runtime\\";

const Test = _Runtime.TypeAliasDeclaration(\\"Test\\", function (typeFactory) {
  const type = _Runtime.NullKeyword;
  return type;
}, []);"
`;

exports[`Keywords NumberKeyword 1`] = `
"import _Runtime from \\"@manta-style/runtime\\";

const Test = _Runtime.TypeAliasDeclaration(\\"Test\\", function (typeFactory) {
  const type = _Runtime.NumberKeyword;
  return type;
}, []);"
`;

exports[`Keywords ObjectKeyword 1`] = `
"import _Runtime from \\"@manta-style/runtime\\";

const Test = _Runtime.TypeAliasDeclaration(\\"Test\\", function (typeFactory) {
  const type = _Runtime.ObjectKeyword;
  return type;
}, []);"
`;

exports[`Keywords StringKeyword 1`] = `
"import _Runtime from \\"@manta-style/runtime\\";

const Test = _Runtime.TypeAliasDeclaration(\\"Test\\", function (typeFactory) {
  const type = _Runtime.StringKeyword;
  return type;
}, []);"
`;

exports[`Keywords UndefinedKeyword 1`] = `
"import _Runtime from \\"@manta-style/runtime\\";

const Test = _Runtime.TypeAliasDeclaration(\\"Test\\", function (typeFactory) {
  const type = _Runtime.UndefinedKeyword;
  return type;
}, []);"
`;
